pytorch_lightning/loops/optimizer/optimizer_loop.py

```diff
@@ -158,16 +158,15 @@ def _run_optimization(
         else:
             self._optimizer_step(optimizer, opt_idx, batch_idx, closure)
 
-        result = closure.get_result()
+        result = closure.result
 
-        if result:
+        if result.loss:
```

Comment:
this should be a None check. The loss is a tensor here 

```suggestion
        if result.loss is not None:
```